consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -297,12 +308,13 @@ impl<T: Payload> BlockStore<T> {
         // corruption, for example.
         match self.get_block(qc.certified_block().id()) {
             Some(executed_block) => {
-                assert_eq!(
-                        executed_block.compute_result().executed_state.state_id,
-                        qc.certified_block().executed_state_id(),
-                        "QC for block {} has a different execution result than the local computation. Restart and try again.",
-                        qc.certified_block().id(),
-                    );
+                ensure!(
```

Comment:
why not use assert_eq?

Reply:
I thought it's safer and if someone carries a wrong genesis QC and may crash the node. Although we shouldn't insert a QC if we already have one, but still being defensive.

The tradeoff is if the local execution result mismatch the quorum, we'll see the node stuck and print a bunch of logs instead of crashing, but I think that's acceptable.

Reply:
oh, I see, its the semantics! crash versus try to identify a safer means to recover...